modules/ossm-migrating-gateways-scenarios.adoc

```diff
@@ -0,0 +1,181 @@
+
+// Module included in the following assemblies:
+//
+// * service-mesh-docs-main/migrating/migrating-gateways/ossm-migrating-gateways-assembly.adoc
+
+:_mod-docs-content-type: PROCEDUR
+[id="ossm-migrating-gateways-scenarios_{context}"]
+= Gateway migration scenarios
+
+There are three scenarios for migrating gateways from {SMProductName} 2 to {SMProduct} 3:
+
+* Gateway canary migration
+* Dedicated application gateway migration
+* Shared gateway migration
+
+Follow the procedure for the scenario that best fits your environment.
+
+== Gateway canary migration
+
+The gateway canary migration scenario is recommend for a gradual rollout. It uses multiple gateway versions with maximum control over gateway rollout.
+
+.Procedure
+
+. Based on your deployment migration method, label the gateway namespace to ensure injection from the new mesh is enabled for the namespace:
```

Comment:
We should just point users to see migration guides for their env to see how to work with injection labels. It will be very difficult to describe all cases here. Can we just say something like: To understand which injection labels should be used for your gateway namespace, visit relevant chapter in migration guide for cluster wide or multitenant.

Reply:
If not we should stick what we have in the GH document:

    Label the gateway namespace to ensure injection from the new mesh is enabled for the namespace (this differs between multitenancy and cluster-wide meshes), ensuring to add the maistra.io/ignore-namespace: "true" label as well as remove istio-injection=enabled if needed.

    NOTE: Look back at the migrating workloads step in your migration guide to understand which labels you should use in your specific migration case and understand why the maistra.io/ignore-namespace: "true" label is needed.